### PR TITLE
LPS-142508 select multiple Categories when creating a navigation menus

### PR DESCRIPTION
**Pull request by lfbesada**
https://issues.liferay.com/browse/LPS-142508

cc: @sandrodw3

### CODE REVIEW

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by lfbesada** [comment]
ci:test:relevant

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/site_navigation_menu_editor/components/AddItemDropdown.js [hunk 1]
```diff
@@ -19,6 +19,43 @@ import React, {useState} from 'react';
...
+	if (value) {
+		delete infoItem.value;
+		infoItem = {...value};
+	}
+
+	infoItem.siteNavigationMenuId = siteNavigationMenuId;
+
+	const namespacedInfoItem = Liferay.Util.ns(portletNamespace, infoItem);
+
+	return namespacedInfoItem;
```
Comment (p2kmgcl) [inline modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/site_navigation_menu_editor/components/AddItemDropdown.js:56]: I think we can just return `Liferay.Util.ns()` here without creating a new variable :smile:

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/site_navigation_menu_editor/components/AddItemDropdown.js [hunk 2]
```diff
@@ -56,9 +56,32 @@ function getNamespacedInfoItem(
...
+	portletNamespace,
+	selectedItems,
+	siteNavigationMenuId
+) {
+	if (!selectedItems.length) {
+		return;
+	}
+
+	const infoItems = {
+		items: JSON.stringify(selectedItems),
```
Comment (p2kmgcl) [inline modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/site_navigation_menu_editor/components/AddItemDropdown.js:69]: Don't we want to call `getNamespacedInfoItem` for each of these?

File: modules/apps/asset/asset-categories-item-selector-web/src/main/resources/META-INF/resources/js/SelectAssetCategoryInfoItem.es.js [hunk 3]
```diff
@@ -110,6 +113,26 @@ function SelectAssetCategory({
 				</ClayLayout.ContainerFluid>
 			</form>
 
+			{!!selectedItemsCount && (
```
Comment (p2kmgcl) [inline modules/apps/asset/asset-categories-item-selector-web/src/main/resources/META-INF/resources/js/SelectAssetCategoryInfoItem.es.js:116]: Some comments about the `!!foo && bar` pattern:

- Instead of `!!foo` prefer using `Boolean(foo)`, which actually uses less operations (only casts the value to boolean) at it is more clear what it's happening.
- You can also avoid using `!!` and `Boolean()` by using a short if instead of an `&&`, something like `foo ? <MyComponent /> : null` will work (it's a bit wordy but we are using it in many places).

Comment (sandrodw3) [reply modules/apps/asset/asset-categories-item-selector-web/src/main/resources/META-INF/resources/js/SelectAssetCategoryInfoItem.es.js:116]: I didn't know Boolean was more optimal 😄
Anyway, I think I prefer the second option you suggested, thanks!

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/site_navigation_menu_editor/components/AddItemDropdown.js (hunk 2 above)
Comment (sandrodw3) [reply modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/site_navigation_menu_editor/components/AddItemDropdown.js:69]: We thought about it, but @lfbesada told me she preferred this format 😛

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/site_navigation_menu_editor/components/AddItemDropdown.js (hunk 1 above)
Comment (sandrodw3) [reply modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/site_navigation_menu_editor/components/AddItemDropdown.js:56]: Done!

**Comment by sandrodw3** [comment]
Frontend review already done 🥰
Closing this one so @lfbesada can work in the backend changes.
Frontend changes done here: https://github.com/lfbesada/liferay-portal/pull/38.
